adapters/faraday/lib/opentelemetry/adapters/faraday/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Faraday
+      module Middlewares
+        class TracerMiddleware < ::Faraday::Middleware
+          def call(env)
+            tracer.in_span(env.url.to_s,
+                          attributes: { 'component' => 'http',
+                                        'http.method' => env.method },
+                          kind: :client) do |span|
+              trace_request(span, env)
+
```

Comment:
We're missing context propagation in the request.

Reply:
I've now added `propogate_context`, but it could use more verification.

Reply:
:heavy_check_mark: